### PR TITLE
Update log injection semantics

### PR DESCRIPTION
**Pull request by RassK**
Updates log injection semantics

| Field | Serilog format | Common format |
| --- | --- | --- |
| Service name | "service_name"  | "service.name" |
| Service version **(new)** | "service_version" | "service.version" | 
| Environment | "deployment_environment" | "deployment.environment" |
| Tracer ID | "trace_id" | "trace_id" |
| Span ID | "trace_id" | "span_id" |

| Log lib | Key syntax | Status |
| --- | --- | --- |
| ILogger | CorrelationIdentifier.Common (originally Serilog) | :heavy_check_mark: |
| Log4Net | CorrelationIdentifier.Common  | :heavy_check_mark: |
| NLog | CorrelationIdentifier.Common | :heavy_check_mark: |
| Serilog | CorrelationIdentifier.Serilog | :heavy_check_mark: |

### CODE REVIEW

**Comment by RassK** [comment]
Log4Net example output

```
2021-11-12 21:32:49,860 [1] INFO LogsInjection.Log4Net.Program {deployment.environment: "(null)", service.name: "(null)", service.version: "(null)", trace_id: "(null)", span_id: "(null)"} - [ExcludeMessage]Entering Datadog scope.
2021-11-12 21:32:50,269 [1] INFO LogsInjection.Log4Net.Program {deployment.environment: "(null)", service.name: "(null)", service.version: "(null)", trace_id: "(null)", span_id: "(null)"} - Calling the PluginApplication.Program in a separate AppDomain
2021-11-12 21:32:51,485 [1] INFO LogsInjection.Log4Net.Program {deployment.environment: "(null)", service.name: "(null)", service.version: "(null)", trace_id: "(null)", span_id: "(null)"} - Returned from the PluginApplication.Program call
2021-11-12 21:32:51,538 [1] INFO LogsInjection.Log4Net.Program {deployment.environment: "(null)", service.name: "(null)", service.version: "(null)", trace_id: "(null)", span_id: "(null)"} - [ExcludeMessage]Exited Datadog scope.
```

File: docs/LOGS_INJECTION.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Versions ≥ 1.2.11 (.NET Framework)
+- Versions ≥ 2.0.12 (.NET core)
+
+Available layouts:
+- JSON format: `SerializedLayout` (from the `log4net.Ext.Json` NuGet package)
+- Raw format: `PatternLayout` (requires manual configuration)
+
+### NLog
+
+- Versions ≥ 1.0.0.505 (.NET Framework)
```
Comment (pjanotti) [inline docs/LOGS_INJECTION.md:36]: On main the minimal version is 4.5 so upstream added support for more down level versions?

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ILoggerTests.cs [hunk 2]
```diff
@@ -24,7 +24,7 @@ public class ILoggerTests : LogsInjectionTestBase
                 FileName = "simple.log",
                 RegexFormat = @"""{0}"":{1}",
                 UnTracedLogTypes = UnTracedLogTypes.EnvServiceTracingPropertiesOnly,
-                PropertiesUseSerilogNaming = true
+                PropertiesUseSerilogNaming = false
```
Comment (pjanotti) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ILoggerTests.cs:27]: I think it is worth a comment here why this different from upstream.

**Review by pjanotti — APPROVED** [review]
LGTM, just a Q and a suggestion for a comment.

File: docs/LOGS_INJECTION.md (hunk 1 above)
Comment (RassK) [reply docs/LOGS_INJECTION.md:36]: seems some of the defined versions they test are not supported, but they are still verifying no correlations for these versions. I need to find the actual minimal versions. So currently these versions are wrong...

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ILoggerTests.cs (hunk 2 above)
Comment (RassK) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ILoggerTests.cs:28]: Comment added.

File: docs/LOGS_INJECTION.md (hunk 1 above)
Comment (RassK) [reply docs/LOGS_INJECTION.md:36]: Looked into this once more, I can see that actually only Serilog is restricted to versions 2+. DD docs seem to confirm this also.

**Pull request merged**